maup/repair.py

```diff
@@ -1,27 +1,1217 @@
+                        first_split.geoms[k], sorted_line_segments[line_pair[1]]
+                    )
+                else:
+                    new_holes.append(first_split.geoms[k])
+
+            # Find the piece in second_split that also touches the first line.
+            # THIS WILL BE OUR PIECE TO ADJOIN TO ONE OF THE TWO POLYGONS!
+
+            for k in range(len(second_split.geoms)):
+                if second_split.geoms[k].intersects(sorted_line_segments[line_pair[0]]):
+                    piece_to_connect = second_split.geoms[k]
+                    poly_pair_found = True  # SUCCESS!!
+                else:
+                    new_holes.append(second_split.geoms[k])
+
+        if poly_pair_found == True:
+            if (
+                piece_to_connect.intersection(
+                    hole_boundaries_df["geometry"][polys_to_connect[0]]
+                ).length
+                > piece_to_connect.intersection(
+                    hole_boundaries_df["geometry"][polys_to_connect[1]]
+                ).length
+            ):
+                poly_to_add_to = hole_boundaries_df["target"][polys_to_connect[0]]
+            elif (
+                piece_to_connect.intersection(
+                    hole_boundaries_df["geometry"][polys_to_connect[0]]
+                ).length
+                < piece_to_connect.intersection(
+                    hole_boundaries_df["geometry"][polys_to_connect[1]]
+                ).length
+            ):
+                poly_to_add_to = hole_boundaries_df["target"][polys_to_connect[1]]
+
+        else:  # Try again with a different polygon pair
+            distance_data_sorted.remove(distance_data_sorted[0])
+
+    if poly_pair_found == False:  # Not much to do at this point except join by shared
+        # perimeter and hope for the best!  (Usually these will
+        # be small enough that they'll get converted to queen
+        # adjacencies eventually anyway.)
+        shared_perimeters = []
+
+        for b_ind in hole_boundaries_df.index:
+            g_ind = hole_boundaries_df["target"][b_ind]
+            shared_perimeters.append(
+                (b_ind, hole_boundaries_df["geometry"][b_ind].length)
+            )
+
+        max_shared_perim = sorted(shared_perimeters, key=lambda tup: tup[1])[-1]
+
+        poly_to_add_to = hole_boundaries_df["target"][max_shared_perim[0]]
+        piece_to_connect = hole
+        new_holes = []
+
+    return poly_to_add_to, piece_to_connect, new_holes
+
+
+# Convert all rook adjacencies with boundary length less than min_rook_length to queen adjacencies
+
+
+def small_rook_to_queen(geometries0_df, min_rook_length):
+
+    geometries_df = geometries0_df.copy()
+
+    adj_df = adjacencies(
+        geometries_df, output_type="geodataframe"
+    )  # We're assuming the input is clean, so these should all be 1-D or less
+
+    for ind in adj_df.index:
+        if adj_df["geometry"][ind].geom_type == "GeometryCollection":
```

Comment:
We have a lot checks like this scattered throughout this function. It would improve overall readability to define some tiny little helpers, e.g. 
```
def _is_collection(geom):
    return geom.geom_type == "GeometryCollection"
```